contracts/modules/TransferManager/VolumeDumpingRestrictionManager.sol

```diff
@@ -0,0 +1,325 @@
+
+        if (_startTime == 0) {
+            _startTime = now;
+        }
+
+        require(_endTime > _startTime, "End time must be greater than start time");
+
+        volumeRestriction[_userAddress] = VolumeRestriction(_percentAllowed, _startTime, _endTime, _rollingPeriod);
+
+        emit ModifyVolumeDumping(
+            _userAddress,
+            _percentAllowed,
+            _startTime,
+            _endTime,
+            _rollingPeriod
+        );
+    }
+    
+    /**
+    * @notice Get volume dumping restriction rules for a user address
+    * @param _userAddress User Address to get the rule
+    */
+    function getVolumeDumpingRestrictions(address _userAddress) public view returns (
+        uint256 _percentAllowed , uint256 _startTime, uint256 _endTime, uint256 _rollingPeriod
+    ) {
+        VolumeRestriction storage userRestriction = volumeRestriction[_userAddress];
+        return (
+            userRestriction.percentAllowed,
+            userRestriction.startTime,
+            userRestriction.endTime,
+            userRestriction.rollingPeriod
+        );
+    }
+
+    /**
+     * @notice This function returns the signature of configure function
+     */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(0);
+    }
+
+
+    /**
+     * @notice Return the permissions flag that are associated with Volume Restriction transfer manager
+     */
+    function getPermissions() public view returns(bytes32[]) {
+        bytes32[] memory allPermissions = new bytes32[](1);
+        allPermissions[0] = VOLUME_RESTRICTION;
+        return allPermissions;
+    }
+
+     /**
+    * @notice Takes a userAddress as input, and returns a uint that represents the number of tokens allowed to be withdrawn right now
+    * @param _userAddress User Address to get the rule
+    * @param _periodId Time period identifier
+    */
+    function _currentAllowedAmount(address _userAddress, uint256 _periodId) internal returns (uint256) {
+        VolumeRestriction storage userDumpingRestriction = volumeRestriction[_userAddress];
+        
+        // get already withdrawn amount in period
+        uint256 alreadyWithdrawn = volumeTally[_userAddress][_periodId];
+        
+        // if the already with drawn is 0
+        // means its essentially a new period
+        if(alreadyWithdrawn == 0){
+            volumePeriodIds[_userAddress].push(_periodId);
+        }
+
+        uint256 currentUserBalance = ISecurityToken(securityToken).balanceOf(_userAddress);
+        // add already withdrawn amount for this period 
+        // to be able to get the correct value
+
+        uint256 percentAllowed = userDumpingRestriction.percentAllowed.mul(uint256(10)**16);
+
+        // totalAmountAllowedForPeriod = currentbalance * 20/100
+        uint256 totalAmountAllowedForPeriod = currentUserBalance.add(alreadyWithdrawn).mul(percentAllowed).div(100 * uint256(10)**16);
```

Comment:
This would be violating the design document shared with me. According to https://www.lucidchart.com/documents/edit/c96ff64c-5b89-4b9c-985a-01637266bf88/0?shared=true&

![image](https://user-images.githubusercontent.com/8148384/47265000-151e3500-d519-11e8-9a05-fdbd6951e592.png)

Reply:
@satyamakgec Can you help confirm this?